File: ruby/leap/leap.rb
class Year

  class << self
    def leap?(year)
      div_by_4?(year) && century_check?(year)
    end


private

    def div_by_4?(year)
      return true if year % 4 == 0
    end

    def century_check?(year)
      return true if year % 100 != 0
      return true if year % 400 == 0
      return false if year % 100 == 0
    end
  end
end

class BookKeeping
  VERSION = 3
end

File: javascript/leap/leap.js
//
// This is only a SKELETON file for the "Leap" exercise. It's been provided as a
// convenience to get you started writing code faster.
//
var year = function(year) {
  this.year = year
};


year.prototype.isLeap = function() {

  if (this.year % 4 === 0 && (this.year % 100 !== 0 || this.year % 400 === 0)) {
    return true
  } else {
    return false
  }
};


module.exports = year;
